exporter/awsemfexporter/metric_translator.go

```diff
@@ -193,87 +262,72 @@ func TranslateCWMetricToEMF(cwMetricLists []*CWMetrics) []*LogEvent {
 	return ples
 }
 
-// Translates OTLP Metric to list of CW Metrics
-func getCWMetrics(metric *pdata.Metric, namespace string, instrumentationLibName string, config *Config) (cwMetrics []*CWMetrics) {
-	var dps DataPoints
-
-	if metric == nil {
-		return
+// Build grouped metric from Datapoint and pdata.Metric
+func buildGroupedMetric (dp DataPoint, pMetricData *pdata.Metric, namespace string) *GroupedMetric {
+	timestamp := time.Now().UnixNano() / int64(time.Millisecond)
+	labels := make(map[string]interface{})
+	metrics := make(map[string]MetricInfo)
+	// Extract labels
+	labelMap := dp.LabelsMap()
+	// v pdata.StringValue?
+	// v.Value()?
+	labelMap.ForEach(func(k string, v string) {
+		labels[k] = v
+	})
+	// Extract metric
+	var metricVal interface{}
+	switch metric := dp.(type) {
+	case pdata.IntDataPoint:
+		metricVal = int64(metric.Value())
+		if needsCalculateRate(pMetricData) {
+			metricVal = calculateRate(labels, metric.Value(), timestamp)
+		}
+	case pdata.DoubleDataPoint:
+		metricVal = float64(metric.Value())
+		if needsCalculateRate(pMetricData) {
+			metricVal = calculateRate(labels, metric.Value(), timestamp)
+		}
+	case pdata.DoubleHistogramDataPoint:
+		bucketBounds := metric.ExplicitBounds()
+		metricVal = &CWMetricStats{
+			Min:   bucketBounds[0],
+			Max:   bucketBounds[len(bucketBounds)-1],
+			Count: metric.Count(),
+			Sum:   metric.Sum(),
+		}
 	}
-
-	// metric measure data from OT
-	metricMeasure := make(map[string]string)
-	metricMeasure["Name"] = metric.Name()
-	metricMeasure["Unit"] = metric.Unit()
-	// metric measure slice could include multiple metric measures
-	metricSlice := []map[string]string{metricMeasure}
-
-	// Retrieve data points
-	switch metric.DataType() {
-	case pdata.MetricDataTypeIntGauge:
-		dps = IntDataPointSlice{metric.IntGauge().DataPoints()}
-	case pdata.MetricDataTypeDoubleGauge:
-		dps = DoubleDataPointSlice{metric.DoubleGauge().DataPoints()}
-	case pdata.MetricDataTypeIntSum:
-		dps = IntDataPointSlice{metric.IntSum().DataPoints()}
-	case pdata.MetricDataTypeDoubleSum:
-		dps = DoubleDataPointSlice{metric.DoubleSum().DataPoints()}
-	case pdata.MetricDataTypeDoubleHistogram:
-		dps = DoubleHistogramDataPointSlice{metric.DoubleHistogram().DataPoints()}
-	default:
-		config.logger.Warn(
-			"Unhandled metric data type.",
-			zap.String("DataType", metric.DataType().String()),
-			zap.String("Name", metric.Name()),
-			zap.String("Unit", metric.Unit()),
-		)
-		return
+	if metricVal == nil {
+		return nil
```

Comment:
Do you need to check the return value `nil` in the function where call this function?

Reply:
yes I think so